apps/admin-portal/src/app/resources/overview/resource-types/recipes/recipe.form.ts

```diff
@@ -47,48 +58,73 @@ export class RecipeForm extends BaseResourceFormComponent {
     protected override getResource(resourceId: string) {
         if (resourceId === TEMP_RESOURCE_ID) return of(TEMP_RECIPE);
         this.isLoading.set(true);
-        return this.recipesService.getById(resourceId);
+        return this.recipesService.getById(resourceId).pipe(map((recipe) => recipe.toDto()));
     }
 
     protected override populateForm() {
-        const { craftingTime, technologyTree, technologyPoints } = this.resourceService.resource() as Recipe;
+        const { craftingTime, technologyTreeId, technologyPoints, inputs, outputs } =
+            this.resourceService.resource() as RecipeDto;
 
         this.form.reset({
             craftingTime: craftingTime,
-            technologyTreeId: technologyTree?.id ?? '',
+            technologyTreeId: technologyTreeId ?? '',
             technologyPoints: technologyPoints,
         });
+
+        this.form.controls.inputs.clear();
+        (inputs ?? []).forEach((dto, index) => {
+            this.form.controls.inputs.insert(index, addRecipeItem(this.environmentInjector, 'input', dto));
+        });
+
+        this.form.controls.outputs.clear();
+        (outputs ?? []).forEach((dto, index) => {
+            this.form.controls.outputs.insert(index, addRecipeItem(this.environmentInjector, 'output', dto));
+        });
     }
 
     protected override onFormChange() {
         const formRecipe = this.createRecipeFromFormValue();
 
-        if (formRecipe.technologyTree?.id !== (this.resourceService.updatedResource() as Recipe)?.technologyTree?.id) {
-            if (formRecipe.technologyTree?.id) {
-                this.technologyTreesService
-                    .getById(formRecipe.technologyTree.id)
-                    .pipe(takeUntilDestroyed(this.destroyRef))
-                    .subscribe({
-                        next: ({ maxPoints }) => {
-                            this.form.controls.technologyPoints.setValidators([
-                                Validators.required,
-                                Validators.min(0),
-                                Validators.max(maxPoints),
-                            ]);
-                            this.form.controls.technologyPoints.updateValueAndValidity();
-                        },
-                    });
-            }
+        if (formRecipe.technologyTreeId && formRecipe.technologyTreeId !== TEMP_RESOURCE_ID) {
+            this.technologyTreesService
+                .getById(formRecipe.technologyTreeId)
+                .pipe(takeUntilDestroyed(this.destroyRef))
+                .subscribe({
+                    next: (technologyTree) => {
+                        this.form.controls.technologyPoints.setValidators([
+                            Validators.required,
+                            Validators.min(0),
+                            Validators.max(technologyTree.maxPoints),
+                        ]);
```

Comment:
When a different technology tree was selected the resource is fetched from the API backend so that we can update maximum allowed points for the technology points input